### PR TITLE
Cleanup: Fix release manifest target

### PR DESCRIPTION
**Pull request by barney-s**
### Change description

Cleanup: Fix release manifest target.

This is to avoid changes in release files everytime some-one tests locally.

- Add a separate make release-manifests target for release to CC
- Rename existing release-* to release-test-* to indicate they are not official release targets
- Generate release-test* manifests in release/test which is .gitignore'd
- Change test files to use manifests from release/test/* folder
- Add separate kustomization target under config/release

### CODE REVIEW

File: experiments/compositions/composition/config/release/manager_auth_proxy_patch.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          limits:
+            cpu: 500m
+            memory: 512Mi
+          requests:
+            cpu: 5m
+            memory: 128Mi
+      - name: manager
+        args:
+        - "--health-probe-bind-address=:8081"
+        - "--metrics-bind-address=127.0.0.1:8080"
```
Comment (cheftako) [inline experiments/compositions/composition/config/release/manager_auth_proxy_patch.yaml:52]: Shouldn't the metrics port also be declared as a container port?

Comment (barney-s) [reply experiments/compositions/composition/config/release/manager_auth_proxy_patch.yaml:52]: This is a copy of the kubebuilder generated patch. I dont normally touch these files. Can check.

Comment (barney-s) [reply experiments/compositions/composition/config/release/manager_auth_proxy_patch.yaml:52]: https://book.kubebuilder.io/reference/metrics.html#protecting-the-metrics

Comment (barney-s) [reply experiments/compositions/composition/config/release/manager_auth_proxy_patch.yaml:52]: https://github.com/brancz/kube-rbac-proxy/#motivation

**Pull request merged**